donkeycar/management/base.py

```diff
@@ -464,6 +464,169 @@ def run(self, args):
 
+class TrainRemote(BaseCommand):
+    """
+    train models on a remote server
+    """
+    # example command: 
+    #     donkey trainremote --tub ~/mycar/data/tub_4_21-06-11/ ~/mycar/data/tub_3_21-06-11/ --path ~/mycar --model modelname
+    #     --url https://hq.robocarstore.com/train/submit_job --get https://hq.robocarstore.com/train/refresh_job_statuses
+
+    WLAN="wlan0"
+    HOTSPOT_IF_NAME="uap0"
+    REFRESH_JOB_STATUS_URL = "https://hq.robocarstore.com/train/refresh_job_statuses"
+    SUBMIT_JOB_URL = "https://hq.robocarstore.com/train/submit_job"
+    TIMEOUT = 900 # limit the training to 15min
+
+    def parse_args(self, args):
+        parser = argparse.ArgumentParser(prog='train', usage='%(prog)s [options]')
+        parser.add_argument('--tub', nargs='+', help='tub data for training')
+        parser.add_argument('--path', default=None, help='path where to create car folder')
+        parser.add_argument('--url', default=None, help='url of the remote server to submit job')
+        parser.add_argument('--get', default=None, help='url of the remote server to get statuses')
+        parser.add_argument('--model', default=None, help='output model name')
+
+        parsed_args = parser.parse_args(args)
+        return parsed_args
+
+    def generate_tub_archive(self, tub_paths, carapp_path):
+        import tempfile
+        import tarfile
+        from pathlib import Path
+        print("generating tub archive")
+        f = tempfile.NamedTemporaryFile(mode='w+b', suffix='.tar.gz', delete=False)
+
+        with tarfile.open(fileobj=f, mode='w:gz') as tar:
+            for tub_path in tub_paths:
+                p = Path(tub_path)
+                tar.add(p, arcname=p.name)
+            tar.add(f"{carapp_path}/myconfig.py", arcname="myconfig.py")
+
+        f.close()
+
+        return f.name
+
+    def get_wlan_mac_address(self, wlan):
+        import netifaces
+
+        interfaces = netifaces.interfaces()
+        if wlan not in interfaces:
+            return "None"
+        addrs =  netifaces.ifaddresses(wlan)
+
+        if addrs is None:
+            return "None"
+
+        if (netifaces.AF_LINK in addrs) and (len(addrs[netifaces.AF_LINK]) == 1):
+            return addrs[netifaces.AF_LINK][0]['addr']
+        else:
+            return "None"
+
+    def submit_job(self, tub_paths, carapp_path, submit_job_url):
+        import requests
+        from requests_toolbelt.multipart.encoder import MultipartEncoder
+        from donkeycar import __version__
+
+        filename = self.generate_tub_archive(tub_paths, carapp_path)
+        mp_encoder = MultipartEncoder(
+            fields={
+                'device_id': self.get_wlan_mac_address(self.WLAN),
+                'hostname' : gethostname(),
+                'tub_archive_file': ('file.tar.gz', open(filename, 'rb'), 'application/gzip'),
+                'donkeycar_version': str(__version__)
+            }
+        )
+        print(f"URL submitted to: {submit_job_url}")
```

Comment:
Please switch to logging statements instead of print, that's what we are using now.